### PR TITLE
Storing versions name

### PR DESCRIPTION
**Pull request by iisuslik43**

### CODE REVIEW

File: src/main/proto/cosmas.proto [hunk 1]
```diff
@@ -1,210 +1,232 @@
...
+    string userName = 4;
+    string versionName = 5;
+}
+
+message FileInfo {
+    string projectId = 1;
+    string fileId = 2;
+}
+
+message VersionsName {
```
Comment (dbarashev) [inline src/main/proto/cosmas.proto]:
```suggestion
message GenerationNameMap {
```

File: src/main/proto/cosmas.proto [hunk 2]
```diff
@@ -1,210 +1,232 @@
...
+    map<int64, string> generationToName = 1;
+}
+
+message Dictionary {
```
Comment (dbarashev) [inline src/main/proto/cosmas.proto]: "Dictionary" says anything but mapping of file id to version names.
Rename to e.g. `FileIdGenerationNameMap`?

File: src/main/proto/cosmas.proto [hunk 3]
```diff
@@ -1,210 +1,232 @@
...
+    string versionName = 5;
+}
+
+message FileInfo {
+    string projectId = 1;
+    string fileId = 2;
+}
+
+message VersionsName {
+    map<int64, string> generationToName = 1;
```
Comment (dbarashev) [inline src/main/proto/cosmas.proto]: with more appropriate message name, this field can be called just `value`

File: src/main/proto/cosmas.proto [hunk 4]
```diff
@@ -1,210 +1,232 @@
...
+}
+
+message Dictionary {
+    map<string, VersionsName> filesToVersionsName = 1;
```
Comment (dbarashev) [inline src/main/proto/cosmas.proto]: same about field name

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt [hunk 5]
```diff
@@ -708,6 +715,47 @@ class CosmasGoogleCloudService(
         return CosmasProto.FileIdChangeMap.parseFrom(mapBytes.getContent()).prevIdsMap
     }
 
+    override fun renameVersion(request: RenameVersionRequest,
+                               responseObserver: StreamObserver<RenameVersionResponse>) = logging(
+            "restoreDeletedFile", request.info.projectId, request.fileId) {
+        val dictionaryName = "${request.info.projectId}-dictionary"
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt]: Any sense to create extension method on `ProjectInfo`?

File: src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt [hunk 6]
```diff
@@ -1621,6 +1732,19 @@ class CosmasGoogleCloudServiceTest {
         return deleteFileRecorder
     }
 
+    private fun renameVersion(generation: Long, name: String, fileId: String = FILE_ID,
+                           info: ProjectInfo = projectInfo()): StreamRecorder<RenameVersionResponse> {
```
Comment (dbarashev) [inline src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt]: something is wrong with the indents

File: src/main/proto/cosmas.proto (hunk 1 above)
Comment (iisuslik43) [reply src/main/proto/cosmas.proto]: ok

File: src/main/proto/cosmas.proto (hunk 3 above)
Comment (iisuslik43) [reply src/main/proto/cosmas.proto]: ok

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt (hunk 5 above)
Comment (iisuslik43) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt]: I m refactoring all this stuff in new pr

File: src/main/proto/cosmas.proto (hunk 4 above)
Comment (iisuslik43) [reply src/main/proto/cosmas.proto]: ok

File: src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt (hunk 6 above)
Comment (iisuslik43) [reply src/test/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudServiceTest.kt]: fixed

File: src/main/proto/cosmas.proto (hunk 2 above)
Comment (iisuslik43) [reply src/main/proto/cosmas.proto]: ok

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt [hunk 7]
```diff
@@ -722,6 +729,47 @@ class CosmasGoogleCloudService(
...
+            "renameVersion", request.info.projectId, request.fileId) {
+        val fileIdGenerationNameMapName = "${request.info.projectId}-fileIdGenerationNameMap"
+        val fileIdGenerationNameMap = try {
+            loadFileIdGenerationNameMap(request.info).toBuilder()
+        } catch (e: StorageException) {
+            handleStorageException(e, responseObserver)
+            return@logging
+        }
+
+        val versionsName = fileIdGenerationNameMap.valueMap
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt]: generationNameMap ?

**Review by dbarashev — APPROVED** [review]
LGTM [1 comment]

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt (hunk 7 above)
Comment (iisuslik43) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt]: done

**Pull request merged**